### DISCUSSION TITLE
Is my folder/file structure set up correctly?

### DISCUSSION THREAD

**Discussion by devjiro76** [expo/router#914] [Q&A] [answered]
The screen what I want is that has tabs & drawers together. and they need be managed in Stack.
In this idea, should I make 3 depth of folder?

folder (stack) : _layou.tsx is only for return <Stack />
folder (drawer): _layout.tsx is only for return <Drawer />
folder (tabs): _layout.tsx is only for return <Tabs>

```
├── (stack)
│   ├── (drawer)
│   │   ├── (tabs)
│   │   │   ├── _layout.tsx
│   │   │   ├── index.tsx --> This screen will be have "Tabs & Drawer"
│   │   └── _layout.tsx
│   └── _layout.tsx
└── _layout.tsx
```

Or could I make only one _layout.tsx with nested code like below?

// app/_layout.tsx
```
<Stack>
  <Drawer>
    <Tabs />
  </Drawer>
</Stack>
```

**Comment by marklawlor** [comment, marked as answer]
There's no correct way to do it - you need to design it in a way that works for you app. Because everyone is going to have a slightly different design. E.g: Will you ever have a screen without tabs, but you still want the drawer present? Ok well they should be separate groups then.

I personally always keep they separated, because I logically think of them as separated.
